tools/pythonpkg/duckdb/experimental/spark/sql/session.py

```diff
@@ -191,9 +190,20 @@ def newSession(self) -> "SparkSession":
         return SparkSession(self._context)
 
     def range(
-        self, start: int, end: Optional[int] = None, step: int = 1, numPartitions: Optional[int] = None
+        self,
+        start: int,
+        end: Optional[int] = None,
+        step: int = 1,
+        numPartitions: Optional[int] = None,
     ) -> "DataFrame":
-        raise ContributionsAcceptedError
+        if numPartitions:
+            raise ContributionsAcceptedError
+
+        if end is None and start >= 0:
```

Comment:
There are no tests that don't trigger this condition
What happens if start is negative

Reply:
Is there a reason we don't just make a function call to the `range` function of duckdb?
through `DuckDBPyConnection.table_function` perhaps?

Reply:
Nice, I didn’t know about this table function, switched to it, thanks o/